From f5ce94cf87cfe247adc2cc61368f9990b6a54372 Mon Sep 17 00:00:00 2001
From: Jason Tibbitts <tibbs@math.uh.edu>
Date: Fri, 20 Jul 2018 19:21:09 -0500
Subject: [PATCH] Python 3.7 compatibility fixes

This fixes the python 3.7 incompatibilities arising from 'async'
becoming a reserved word.

The file lib/solaar/async.py is renamed to asks.py.  I picked the name
because it defines the TaskRunner class and it's the best I could come
up in fifteen seconds.

The async function in solar/ui/__init__.py is renamed to ui_async, and
the various imports of that function are changed to match.

Without this patch it doesn't build at all.  I am running with this
patch applied and everything appears to work as expected.

Signed-off-by: Jason Tibbitts <tibbs@math.uh.edu>
---
 lib/solaar/{async.py => tasks.py} | 0
 lib/solaar/ui/__init__.py         | 4 ++--
 lib/solaar/ui/config_panel.py     | 2 +-
 lib/solaar/ui/window.py           | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)
 rename lib/solaar/{async.py => tasks.py} (100%)

diff --git a/lib/solaar/async.py b/lib/solaar/tasks.py
similarity index 100%
rename from lib/solaar/async.py
rename to lib/solaar/tasks.py
diff -ruN Solaar-0.9.2/lib/solaar/ui/__init__.py Solaar-0.9.2-b/lib/solaar/ui/__init__.py
--- Solaar-0.9.2/lib/solaar/ui/__init__.py	2013-07-24 11:40:54.000000000 +0200
+++ Solaar-0.9.2-b/lib/solaar/ui/__init__.py	2018-07-22 17:47:09.739229236 +0200
@@ -123,7 +123,7 @@
 
 del Thread
 
-def async(function, *args, **kwargs):
+def ui_async(function, *args, **kwargs):
 	task = (function, args, kwargs)
 	_task_queue.put(task)
 
@@ -149,7 +149,7 @@
 	def _shutdown(app):
 		# stop the async UI processor
 		_queue_processor.alive = False
-		async(None)
+		ui_async(None)
 
 		tray.destroy()
 		notify.uninit()
diff --git a/lib/solaar/ui/config_panel.py b/lib/solaar/ui/config_panel.py
index 75f9a2c..151b61c 100644
--- a/lib/solaar/ui/config_panel.py
+++ b/lib/solaar/ui/config_panel.py
@@ -23,7 +23,7 @@ from gi.repository import Gtk, GLib
 from threading import Timer as _Timer
 
 from solaar.i18n import _
-from solaar.ui import async as _ui_async
+from solaar.ui import ui_async as _ui_async
 from logitech_receiver.settings import KIND as _SETTING_KIND
 
 #
diff --git a/lib/solaar/ui/window.py b/lib/solaar/ui/window.py
index 6e27e09..8697fc7 100644
--- a/lib/solaar/ui/window.py
+++ b/lib/solaar/ui/window.py
@@ -31,7 +31,7 @@ from solaar import NAME
 from solaar.i18n import _
 from gettext import ngettext
 # from solaar import __version__ as VERSION
-from solaar.ui import async as _ui_async
+from solaar.ui import ui_async as _ui_async
 from logitech_receiver import hidpp10 as _hidpp10
 from logitech_receiver.common import NamedInts as _NamedInts, NamedInt as _NamedInt
 from logitech_receiver.status import KEYS as _K
-- 
2.18.0

